src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,72 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  for (int i = 0; TRUE; i++)
+    {
+      gsize len;
+      g_autoptr (GError) error = NULL;
+      g_autofree gchar *path = g_strdup_printf ("/sys/class/hwmon/hwmon%d/name", i);
+      g_autofree gchar *name = NULL;
+
+      if (!g_file_get_contents (path, &name, &len, &error))
+        {
+          if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+              g_message ("error loading name: %s", error->message);
+
+          break;
+        }
+
+      g_strchomp(name);
+
+      if (!g_str_equal (name, "coretemp") && !g_str_equal (name, "cpu_thermal") &&
+          !g_str_equal (name, "k8temp") && !g_str_equal (name, "k10temp"))
+        {
+          // doesn't match any supported CPU
+          continue;
+        }
+
+      for (int j = 1; TRUE; j++)
```

Comment:
I'm not a fan of this nested loop structure, for a couple of reasons.

One reason is that it makes it makes it complicated to understand what is supposed to happen in case we hit an error when reading the temperature file of a particular hwmon device.  I think the intent here is that we log the error, clear it, and then go back to searching for more temperature devices?  Is it expected that there are multiple matching temperature devices on the system?

That brings to the second reason: in the event that there are multiple matching temperature devices on the system, we'll set the temperature property on the samples object twice.

This function is also just quite long.

Depending on if it's expected that we might have multiple matching devices, I recommend a structure where you first determine the device to use, and then query it.

You might split that into 3 or 4 functions, as so:
 - the read-a-possibly-non-existing-file helper function
 - a function which finds the hwmon device that we want to use
 - a function to read the temperature out of that device
 - a main function that ties it all together

Otherwise if we do intend to query multiple devices, in case one of them fails, it might be nice to express that logic plainly in the main function, and rather have a simple helper function for deciding "is this a valid hwmon device?"

Reply:
> That brings to the second reason: in the event that there are multiple matching temperature devices on the system, we'll set the temperature property on the samples object twice.

Yes, multiple matching devices are expected and sample object is always unique.

Sample for intel and amd looks like this:
`hwmon(NUM) (label)`
where `NUM` is number of current hwmon directory and `label` is unique label in current directory.
As arm doesn't use labels I create a unique one using the counter from second loop.

> This function is also just quite long.

This is true, I'll try to refactor it and split into multiple functions as you suggest.

Thanks for the review!